Source/Shared/Cache.swift

```diff
@@ -27,8 +27,17 @@ public final class Cache<T: Cachable>: BasicHybridCache {
    - Parameter key: Unique key to identify the object in the cache
    - Parameter completion: Completion closure returns object or nil
    */
-
   public func object(_ key: String, completion: @escaping (_ object: T?) -> Void) {
     super.object(forKey: key, completion: completion)
   }
+  
+  /**
+   Tries to retrieve the cache entry from to the front and back cache storages.
+   
+   - Parameter key: Unique key to identify the cache entry in the cache
+   - Parameter completion: Completion closure returns cache entry or nil
+   */
+  public func cacheEntry(_ key: String, completion: @escaping (_ object: CacheEntry<T>?) -> Void) {
```

Comment:
@vadymmarkov I made it public because I wanted a method that would return, not just the object, but object and metadata; I have code that needs to read the expiry time of a cached object (that hasn't expired) and make decisions about how it can be used based on how old it is. Does this sound like a use case you guys want to support? 

I wasn't sure if we should add a new method to the `Cache` public interface or have an overload for the `object` method that takes a `completion` lambda with a `CacheEntry` parameter. The side effect of that is we lose a bit of the swift inference when passing a lambda to the `object` method; users will have to be explicit with their parameter types which means a breaking change. Hence the new method.